### PR TITLE
Fix #892

### LINKED ISSUE #892: RDKit stereo enumeration does not flip existing stereochemistry
**Describe the bug**

Attempting to enumerate all stereoisomers of a molecule with already defined stereochemistry yields no stereoisomers when using the RDKit backend.

**To Reproduce**

Using the toolkit:

```
>>> off_molecule = Molecule.from_smiles('[H][C@@](F)(Cl)Br')
>>>
>>> print(len(off_molecule.enumerate_stereoisomers(
>>>     undefined_only=False, toolkit_registry=RDKitToolkitWrapper())))
>>> 
>>> print(len(off_molecule.enumerate_stereoisomers(
>>>     undefined_only=False, toolkit_registry=OpenEyeToolkitWrapper())))

0
1
```

comparison with the RDKit toolkit directly:

```
>>> rdkit_molecule = Chem.MolFromSmiles('[H][C@@](F)(Cl)Br')
>>> options = StereoEnumerationOptions(onlyUnassigned=False)
>>>
>>> for isomer in EnumerateStereoisomers(rdkit_molecule, options=options):
>>>     print(Chem.MolToSmiles(isomer))

F[C@H](Cl)Br
F[C@@H](Cl)Br
```

**Output**
The full error message (may be large, that's fine. Better to paste too much than too little.)

**Computing environment (please complete the following information):**
 - Operating system: OSX
 - Output of running `conda list` :

```
rdkit                     2020.09.3        py38h4648dd3_0    conda-forge
openeye-toolkits          2020.2.0                 py38_0    openeye
openff-toolkit            0.9.1           dev_0    <develop>
```

**Additional context**

### PR DESCRIPTION
**Pull request by SimonBoothroyd**
Fixes #892.

- [X] Tag issue being addressed
- [X] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/master/openff/toolkit/tests)
- [X] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/master/docs), if applicable
- [X] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [X] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/master/docs/releasehistory.rst)

### CODE REVIEW

File: openff/toolkit/tests/test_molecule.py [hunk 1]
```diff
@@ -1844,6 +1844,62 @@ def test_enumerating_stereo_options(self, toolkit_class):
...
+
+        # test undefined only
+        mol = Molecule.from_smiles(
+            smiles, toolkit_registry=toolkit, allow_undefined_stereo=True
+        )
+        stereoisomers = mol.enumerate_stereoisomers(
+            undefined_only=undefined_only, rationalise=False
+        )
+
+        expected = {
```
Comment (j-wags) [inline openff/toolkit/tests/test_molecule.py:1890]:
```suggestion
        # Ensure that the results of the enumeration are what the test expects.
        # This roundtrips the expected output from SMILES --> OFFMol --> SMILES,
        # since the SMILES for stereoisomers generated in this test may change depending
        # on which cheminformatics toolkit is used.
        expected = {
```

**Review by j-wags — APPROVED** [review]
Beautiful. The investigation in the linked issue was excellent. And the test added in this PR was nicely expansive and gracefully handled every edge case I could think of. Thanks, @SimonBoothroyd
